src/pages/constructor/constructor.js

```diff
@@ -0,0 +1,238 @@
+import React, { useCallback, useEffect, useRef } from 'react';
+import { FormControl, FormHelperText, NativeSelect } from '@material-ui/core';
+import mergeImages from 'merge-images';
+import { useDispatch, useSelector } from 'react-redux';
+import { useStyles } from './constructor.style';
+import { CONSTRUCTOR_TITLES } from '../../translations/constructor.translations';
+import {
+  getConstructorModelById,
+  getModelForConstructor,
+  setModelLoading
+} from '../../redux/constructor/constructor-model/constructor-model.actions';
+import { getConstructorBasic } from '../../redux/constructor/constructor-basic/constructor-basic.actions';
+import { getConstructorPattern } from '../../redux/constructor/constructor-pattern/constructor-pattern.actions';
+import { getConstructorBottom } from '../../redux/constructor/constructor-bottom/constructor-bottom.actions';
+import { getConstructorFrontPocket } from '../../redux/constructor/constructor-front-pocket/constructor-front-pocket.actions';
+import Loader from '../../components/loader';
+import { selectConstructor } from '../../redux/selectors/multiple.selectors';
+
+const map = require('lodash/map');
+
+const { MODEL, BASIC, PATTERN, BOTTOM } = CONSTRUCTOR_TITLES[0];
+
+const useConstructor = () => {
```

Comment:
Extract this hook to a separate file called `hooks.js` near to this one, so we would support single responsibility principle per file

Reply:
fixed